src/test/java/com/revature/studyforce/flashcard/service/AnswerServiceTest.java

```diff
@@ -0,0 +1,121 @@
+import org.springframework.test.context.TestPropertySource;
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * Test class for AnswerService {@link AnswerService}
+ * @author Edson Rodriguez
+ */
+@SpringBootTest
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+@TestPropertySource(locations = "classpath:test-application.properties")
+class AnswerServiceTest {
+
+    @MockBean
+    private UserRepository userRepository;
+
+    @MockBean
+    private FlashcardRepository flashcardRepo;
+
+    @MockBean
+    private AnswerRepository answerRepository;
+
+    @Autowired
+    private AnswerService answerService;
+
+    @Test
+    void givenAnswer_whenCreateAnswer_shouldReturnAnswerDTO(){
+        User user = new User(0,"edson@revature.com","password","Edson","Rodriguez",true,false,false, Authority.USER, Timestamp.valueOf(LocalDateTime.now()),Timestamp.valueOf(LocalDateTime.now()));
+        Flashcard flashcard = new Flashcard(0,user,null,"how is your day",1,1,Timestamp.valueOf(LocalDateTime.now()),null,false);
+        Answer a = new Answer(0,user,flashcard,"tcs filename",0,false,false,Timestamp.valueOf(LocalDateTime.now()), null);
+        AnswerDTO aDTO = new AnswerDTO(0,0,"tcs filename");
+
+        Mockito.when(flashcardRepo.findById(0)).thenReturn(Optional.of(flashcard));
+        Mockito.when(userRepository.findById(0)).thenReturn(Optional.of(user));
+        Mockito.when(answerRepository.save(org.mockito.ArgumentMatchers.isA(Answer.class))).thenReturn(a);
+
+        Answer res = answerService.createAnswer(aDTO);
+
+        assertNotNull(res);
+        assertEquals(0,res.getAnswerId());
+        assertEquals(user,res.getCreator());
+        assertEquals(flashcard,res.getFlashcard());
+        assertEquals("tcs filename",res.getAnswerText());
+        assertEquals(0,res.getAnswerScore());
+        assertFalse(res.isSelectedAnswer());
+        assertFalse(res.isTrainerSelected());
+        assertNotNull(res.getCreationTime());
+        assertNull(res.getResolutionTime());
+    }
+
+    @Test
+    void givenAnswerId_whenDeleteAnswerById_shouldDeleteAnswer(){
+        answerService.deleteAnswerById(1);
+        Mockito.verify(answerRepository, Mockito.times(1)).deleteById(1);
+    }
+
+    @Test
+    void givenFlashcardId_whenGetAllByFlashcardId_shouldReturnPageOfAnswersWithPagination(){
+        List<Answer> answers = new ArrayList<>();
+        User user = new User(0,"edson@revature.com","password","Edson","Rodriguez",true,false,false, Authority.USER, Timestamp.valueOf(LocalDateTime.now()),Timestamp.valueOf(LocalDateTime.now()));
+        User user2 = new User(1,"edson2@revature.com","password2","Edson2","Rodriguez2",true,false,false, Authority.USER, Timestamp.valueOf(LocalDateTime.now()),Timestamp.valueOf(LocalDateTime.now()));
```

Comment:
Using outdated user model, please correct to user model with 1 name field and no password field